### PR TITLE
Plugin of restrict user registration

### PR DESCRIPTION
**Pull request by MeghaBiranje**

### CODE REVIEW

File: restrict_registration/language/en-GB/en-GB.plg_user_restrictregistration.ini [hunk 1]
```diff
@@ -0,0 +1,4 @@
+PLG_USER_RESTRICT_REGISTRATION="Restrict user registration"
```
Comment (manojLondhe) [inline restrict_registration/language/en-GB/en-GB.plg_user_restrictregistration.ini:1]: add copyright

File: restrict_registration/language/en-GB/en-GB.plg_user_restrictregistration.sys.ini [hunk 2]
```diff
@@ -0,0 +1,2 @@
+;__DEPLOY_VERSION__
```
Comment (manojLondhe) [inline restrict_registration/language/en-GB/en-GB.plg_user_restrictregistration.sys.ini:1]: add copyright

File: restrict_registration/restrictRegistration.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * @package     Joomla.Plugin
+ * @subpackage  User.restrictRegistration
+ *
+ * @copyright   Copyright (C) 2005 - 2021 Open Source Matters, Inc. All rights reserved.
```
Comment (manojLondhe) [inline restrict_registration/restrictRegistration.php:6]: update copyright

File: restrict_registration/restrictRegistration.php [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
```
Comment (manojLondhe) [inline restrict_registration/restrictRegistration.php:7]: add license link, as there is no license.ttxt

File: restrict_registration/restrictRegistration.php [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+defined('_JEXEC') or die;
+
+use Joomla\CMS\Language\Text;
+use Joomla\CMS\Plugin\CMSPlugin;
+use Joomla\Utilities\ArrayHelper;
+use Joomla\CMS\MVC\Model\BaseDatabaseModel;
+
+// Load language file for plugin.
+$lang = JFactory::getLanguage();
+$lang->load('plg_user_restrictregistration', JPATH_ADMINISTRATOR);
```
Comment (manojLondhe) [inline restrict_registration/restrictRegistration.php:19]: AFAIK not needed when you add

protected $autoloadLanguage = true;

File: restrict_registration/language/en-GB/en-GB.plg_user_restrictregistration.ini [hunk 6]
```diff
@@ -0,0 +1,4 @@
...
+PLG_USER_RESTRICT_REGISTRATION_USER_COUNT_LABLE="User Restrict Count"
+PLG_USER_RESTRICT_REGISTRATION_USER_COUNT_LABLE_DESC="Enter user restrict count"
```
Comment (manojLondhe) [inline restrict_registration/language/en-GB/en-GB.plg_user_restrictregistration.ini:3]: PLG_USER_RESTRICT_REGISTRATION_USER_COUNT_LABLE="Max Allowed Registrations Count"
PLG_USER_RESTRICT_REGISTRATION_USER_COUNT_LABLE_DESC="Set count of max no. of users that can be registered on website"
PLG_USER_RESTRICT_REGISTRATION_MSG="Cannot add more users. Max no. of allowed registrations count is reached."

**Review by manojLondhe — CHANGES_REQUESTED** [review]

File: restrict_registration/restrictRegistration.php (hunk 3 above)
Comment (MeghaBiranje) [reply restrict_registration/restrictRegistration.php]: done

File: restrict_registration/language/en-GB/en-GB.plg_user_restrictregistration.ini (hunk 6 above)
Comment (MeghaBiranje) [reply restrict_registration/language/en-GB/en-GB.plg_user_restrictregistration.ini]: updated

File: restrict_registration/language/en-GB/en-GB.plg_user_restrictregistration.ini [hunk 7]
```diff
@@ -0,0 +1,11 @@
+;@package     Joomla.Plugin
+;@subpackage  User.restrictRegistration
+;@copyright   Copyright (C) 2020 - 2021 Open Source Matters, Inc. All rights reserved.
```
Comment (manojLondhe) [inline restrict_registration/language/en-GB/en-GB.plg_user_restrictregistration.ini:3]: add for TJ

File: restrict_registration/language/en-GB/en-GB.plg_user_restrictregistration.ini [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+;@license     http://www.gnu.org/copyleft/gpl.html GNU/GPL, see LICENSE.php
```
Comment (manojLondhe) [inline restrict_registration/language/en-GB/en-GB.plg_user_restrictregistration.ini:4]: add link for gpl2

File: restrict_registration/language/en-GB/en-GB.plg_user_restrictregistration.ini [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+
+PLG_USER_RESTRICT_REGISTRATION="Restrict user registration"
+PLG_USER_RESTRICT_REGISTRATION_USER_COUNT_LABLE="Max Allowed Registrations Count"
+
+PLG_USER_RESTRICT_REGISTRATION_USER_COUNT_LABLE_DESC="Set count of max no. of users that can be registered on website"
+
```
Comment (manojLondhe) [inline restrict_registration/language/en-GB/en-GB.plg_user_restrictregistration.ini:10]: remove extra lines

**Pull request merged**